src/Runtime.hs

```diff
@@ -54,28 +54,23 @@ data Params = Params { weight :: Double,
                        annotations :: [[Annotation]]
                      } -- deriving (Eq, Show) -- TODO derive Show instance
 
--- State of the world
+-- State of the runtime system
 data State = State { objs :: [Obj]
                    , constrs :: [C.SubConstr]
-                   , comps :: [ObjComp]
-                   , down :: Bool -- left mouse button is down (dragging)
+                   , comps :: forall a. (Autofloat a) => [ObjComp a]
                    , rng :: StdGen -- random number generator
                    , autostep :: Bool -- automatically step optimization or not
                    , params :: Params
                    }  deriving (Typeable)
 
-type Config = M.Map String S.Expr
-
 -- | Datatypes for computation. ObjComp is gathered in pre-compilation and passed to functions that evaluate the computation.
--- | object name, function name, list of args (TODO resolve them WRT pattern matching)
-data ObjComp = ObjComp { oName :: Name, -- "A"
-                     oProp :: Name, -- "radius"
-                     fnName :: Name, -- computeRadius
-                     fnParams :: [S.Expr] } -- (1.2, B)
-               deriving (Show, Typeable)
-
--- | fn name, list of args (returned by queryCondig)
-type CompInfo = (Name, [S.Expr])
+-- | object name, function name, list of args
+data ObjComp a = ObjComp {
+    oName    :: Name,      -- "A"
```

Comment:
I think this `ObjComp` thing (which I originally put in for property-setting like `prop = f(x)`) is better as a suspended `set` (just like `A.shape.prop` has become a suspended `get`). That will unify our approaches